src/app.js

```diff
@@ -1,14 +1,115 @@
-const http = require('http');
-const logger = require('./logger')
-const config = require('./config')
+const express = require('express')
+const { 
+  createReadStream, 
+  createWriteStream
+} = require('fs')
+const { 
+  unlink, 
+  rename,
+  appendFile 
+} = require('fs').promises
+const readline = require('readline');
+const path = require('path')
+const csv = require('csvtojson');
+const Filter = require('./filter')
+const { port } = require('./config')
 
-const port = config.port ? config.port : 3000;
-const env = config.env ? config.env : 'test';
+const dataFile = path.join(__dirname, '../calendar-data.csv')
+const tempFile = path.join(__dirname, '../calendar-data.tmp')
 
-const server = http.createServer((req, res) => {
-    res.statusCode = 200;
-    res.setHeader('Content-Type', 'application/json');
-    res.end(`Hello world! (ENV: ${env})`);
-});
+const app = express()
+app.use(express.json())
 
-server.listen(port, () => logger.log('Listening on port %s...', port));
+app.get('/events', (req, res) => {
+  const readStream = createReadStream(dataFile)
+
+  const filter = new Filter()
+  const { location } = req.query
+  if (location) {
+    filter.predicate = (it) => JSON.parse(it).location === location
+  }
+    
+  readStream 
+    .pipe(csv())
+    .pipe(filter)
+    .pipe(res)
+})
+
+app.get('/events/:eventId', (req, res) => {
+  const readStream = createReadStream(dataFile)
+
+  const filter = new Filter((it) => JSON.parse(it).id === req.params.eventId)
+
+  readStream 
+    .pipe(csv())
+    .pipe(filter)
+    .pipe(res)
+})
+
+app.post('/events', async (req, res) => {
+  await appendFile(dataFile, Object.values(req.body).join(',') + '\n')
+  res.sendStatus(201)
+})
+
+app.put('/events/:eventId', (req, res) => {
+  const readStream = createReadStream(dataFile)
+  const writeStream = createWriteStream(tempFile)
```

Comment:
Since we are using the same `tempFile` for all requests we might face a situation when two parallel requests are concurrently writing to the same `tempFile` and conflicting with each other.